File: Backend/task/src/main/java/edu/personnel/taskmanagement/task/dto/PageResponse.java
package edu.personnel.taskmanagement.task.dto;

import lombok.*;

import java.util.List;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PageResponse<T> {
    private List<T> content;
    private Integer number;
    private Integer size;
    private Long totalElements;
    private Integer totalPages;
    private Boolean first;
    private Boolean last;
}

File: Frontend/task/src/app/models/PageResponse.ts
export interface PageResponse<T> {
  content: T[]; // The list of items (content)
  number: number; // Current page number (starting from 0)
  size: number; // Number of items per page
  totalElements: number; // Total number of items across all pages
  totalPages: number; // Total number of pages
  first: boolean; // Indicates if it's the first page
  last: boolean; // Indicates if it's the last page
}
